lighthouse-core/test/lib/web-inspector-test.js

```diff
@@ -21,6 +21,21 @@ describe('Web Inspector lib', function() {
     assert.ok(WebInspector.Color);
   });
 
+  it('does not destroy setImmediate', done => {
```

Comment:
not immediately clear what's being tested here. Is this checking for the specific WebInspector (bad) `setImmediate` implementation?

Reply:
yes

`does not break setImmediate`
`does not polyfill setImmediate incorrectly`

any better?

Reply:
yeah, that helps, though maybe a comment about why `arg.foo();` would throw, too :)

Reply:
also TIL `setImmediate` also exists on `require('timers').setImmediate`, so could check `setImmediate === require('timers').setImmediate` in addition to this functional test.

In theory someone could overwrite that as well, but they may deserve to screw things up by the time they reach that level of effort.